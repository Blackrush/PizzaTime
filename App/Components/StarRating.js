import React from 'react'

import StarRating from 'react-native-star-rating'

export default (props) => {
  const starSize =
    props.size == 'medium' ? 11 :
    props.size == 'large'  ? 27 :
      11;

  return (
    <StarRating selectedStar={(rating) => false} disabled
                {...props}
                starSize={starSize}
                starColor="#F9D53B"/>
  )
}