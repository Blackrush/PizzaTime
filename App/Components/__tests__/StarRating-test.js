import React from 'react'
import StarRating from '../StarRating'
import { create as render } from 'react-test-renderer'

test('StarRating renders correctly', () => {
  const tree = render(
    <StarRating starSize="large" rating={8.5} maxStars={10}/>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
