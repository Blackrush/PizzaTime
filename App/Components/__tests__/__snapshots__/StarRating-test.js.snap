exports[`test StarRating renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between"
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F9D53B",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "gray",
            "fontFamily": "FontAwesome",
            "fontSize": 11,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
  </View>
</View>
`;
